### PR TITLE
JETUT-111 JETUT-115 Explain the open id connect authentication mechanism and LDAP identity store

### PR DESCRIPTION
**Pull request by arjantijms**

### CODE REVIEW

File: src/main/antora/modules/security/partials/declare-security-constraints.adoc [hunk 1]
```diff
@@ -0,0 +1,28 @@
+Next we'll define the security constraints in `web.xml`, which tell the security system that access to a given URL or URL pattern is protected, and hence authentication is required:
+
+[source,xml]
+----
+<?xml version="1.0" encoding="UTF-8"?>
+<web-app version="5.0"
```
Comment (BalusC) [inline src/main/antora/modules/security/partials/declare-security-constraints.adoc:6]: Shouldn't this be 6.0?

Comment (arjantijms) [reply src/main/antora/modules/security/partials/declare-security-constraints.adoc:6]: Indeed!

File: src/main/antora/modules/security/pages/security.adoc [hunk 2]
```diff
@@ -1496,3 +1307,411 @@ The test sends a request here to the protected resource along with the headers w
...
+
+This identity store is set to `PROVIDE_GROUPS` (1) only, which means the default `IdentityStoreHandler` will consult this identity store for groups after another identity store has successfully validated the credentials. For our example here we create a `Map` (2) with as key the caller principal name, and as value the set of groups. When the `IdentityStoreHandler` comes asking for the groups (3) of caller "user", a set with just the group "user" is returned.
+
+NOTE: As validation of the `IdentityToken` that's returned by the OpenID Connect Provider is integral to the OpenID Connect flow and not application specific, developers don't have to provide or define an identity store explicitly for this. Such a store is provided by Jakarta Security as an implementation detail, and automatically activated when the OpenID Connect authentication mechanism is activated.
+
+include::partial$identity-store-installed.adoc[]
+
+==== Install and configure Mitre
+
+Installing and configuring Mitre is outside the scope of Jakarta Security itself, but for completeness sake we'll briefly
```
[the remark this answers is not in the document]
Comment (arjantijms) [reply src/main/antora/modules/security/pages/security.adoc:1402]: It is a black box in a way, so not explaining all its native configuration, but on the other hand its also the touch point for the configuration items in de Jakarta Security annotation.

File: src/main/antora/modules/security/pages/security.adoc [hunk 3]
```diff
@@ -771,24 +615,25 @@ public class RestFormAuthCustomStoreIT extends ITBase {
...
 
 You'll learn how to:
 
-. Define security constraints
-. Use the provided Basic authentication mechanism
-. Use the provided database identity store
-. Create a custom identity store
-. Create a custom identity store handler
-. Use the Jakarta Security `SecurityContext`
+. https://jakarta.ee/specifications/servlet/6.0/jakarta-servlet-spec-6.0.html#specifying-security-constraints[Define security constraints,role=external,window=_blank]
```
[the remark this answers is not in the document]
Comment (arjantijms) [reply src/main/antora/modules/security/pages/security.adoc]: I'll replace a few of the most occurring ones.

**Review by kito99 — APPROVED** [review]

**Pull request merged**